### PR TITLE
Add test for ensuring collabdocs initialize empty

### PR DESCRIPTION
**Pull request by mrvisser**
As per @nicolaasmatthijs 's comment in #1130

### CODE REVIEW

**Comment by nicolaasmatthijs** [comment]
@mrvisser: It looks like there was a build failure here:

```
1) Activity push Subscribing verify subscribing and authorization on notification streams:
     Uncaught TypeError: Cannot read property 'restContext' of undefined
      at /home/travis/build/oaeproject/Hilary/node_modules/oae-activity/tests/test-push.js:306:44
      at /home/travis/build/oaeproject/Hilary/node_modules/oae-tests/lib/util.js:9:11133
      at Request._callback (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/lib/util.js:9:8518)
      at Request.self.callback (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/node_modules/request/request.js:121:22)
      at Request.emit (events.js:98:17)
      at Request.<anonymous> (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/node_modules/request/request.js:978:14)
      at Request.emit (events.js:117:20)
      at IncomingMessage.<anonymous> (/home/travis/build/oaeproject/Hilary/node_modules/oae-rest/node_modules/request/request.js:929:12)
      at IncomingMessage.emit (events.js:117:20)
      at _stream_readable.js:938:16
```

Whilst this doesn't look related to this PR, it also doesn't look like the type of error that would indicate an intermediate test failure. Any idea?

File: node_modules/oae-content/tests/test-collabdoc.js [hunk 1]
```diff
@@ -489,6 +489,22 @@ describe('Collaborative documents', function() {
     });
 
     /**
+     * Test that verifies a collabdoc is created and initialized with no content
```
Comment (nicolaasmatthijs) [inline node_modules/oae-content/tests/test-collabdoc.js]: `Test that verifies that`, just to be fully consistent

**Comment by nicolaasmatthijs** [comment]
Left a question about the test failure. Also added a superminor comment. Back to @mrvisser

**Comment by mrvisser** [comment]
> Whilst this doesn't look related to this PR, it also doesn't look like the type of error that would indicate an intermediate test failure. Any idea?

Maybe a race condition related to when `restContext` gets set in the connection life-cycle. I'll investigate.

**Comment by mrvisser** [comment]
```javascript
        /**
         * Test that verifies subscribing and authorization on notification streams
         */
        it('verify subscribing and authorization on notification streams', function(callback) {
            TestsUtil.generateTestUsers(camAdminRestContext, 2, function(err, users, mrvisser, simon) {

                RestAPI.User.getMe(mrvisser.restContext, function(err, mrvisserMeData) {
                    assert.ok(!err);

                    var data = {
                        'authentication': {
                            'userId': mrvisserMeData.id,
                            'tenantAlias': mrvisserMeData.tenant.alias,
                            'signature': mrvisserMeData.signature
                        }
                    };
```

and `RestAPI.User.getMe(mrvisser.restContext, function(err, mrvisserMeData) {` is the line it happens on. I'm assuming `err` was set for some reason, but I can't think of why. For now, I will put an `assert.ok(!err)` in there, but a local run of the tests succeed.

**Comment by mrvisser** [comment]
If it ever bites us on `master` we'll have the logs to work with in that case.

**Comment by mrvisser** [comment]
Followed up. Let's have a look at what the tests say. Assigning back to @nicolaasmatthijs in the meantime.

**Comment by nicolaasmatthijs** [comment]
Thanks for the follow-up. Merged.
